libgnucash/app-utils/gnc-option-date.hpp

```diff
@@ -0,0 +1,180 @@
+ * will be made concrete as the same day as today in the previous month.
+ *
+ * @param period The Relative Date Period to check.
+ * @return true if the date is stand-alone.
+ */
+bool gnc_relative_date_is_single(RelativeDatePeriod);
+
+/**
+ * Report whether the relative date represents the beginning of a date
+ * range. For example START_LAST_MONTH is the beginning of a range.
+ *
+ * @param period The Relative Date Period to check.
+ * @return true if the date is the beginning of a date range
+ */
+bool gnc_relative_date_is_starting(RelativeDatePeriod);
+
+/**
+ * Report whether the relative date represents the end of a date range. For
+ * example END_LAST_MONTH is the end of a range.
+ *
+ * @param period The Relative Date Period to check.
+ * @return true if the date is the end of a date range.
+ */
+bool gnc_relative_date_is_ending(RelativeDatePeriod);
+
+/**
+ * Provide the string representation of a relative date for persisting the
+ * value. This string is not localizable.
+ *
+ * @param period The relative date period.
+ * @return A constant string or nullptr if the period is ABSOLUTE. The string's
+ * lifetime will be that of the Relative Date Period. It must not be freed and
+ * should be copied if the period might be destroyed before the using code is
+ * finished.
+ */
+const char* gnc_relative_date_storage_string(RelativeDatePeriod);
+
+/**
+ * Provide the string representation of a relative date for displaying
+ * value to a user. This string is not localizable.
+ *
+ * @param period The relative date period.
+ * @return A constant string or nullptr if the period is ABSOLUTE. The string's
+ * lifetime will be that of the Relative Date Period. It must not be freed and
+ * should be copied if the period might be destroyed before the using code is
+ * finished.
+ */
+const char* gnc_relative_date_display_string(RelativeDatePeriod);
+
+/**
+ * Provide the description of a relative date. This string is localizable.
+ *
+ * @param period The relative date period.
+ * @return A constant string or nullptr if the period is ABSOLUTE. The string's
+ * lifetime will be that of the Relative Date Period. It must not be freed and
+ * should be copied if the period might be destroyed before the using code is
+ * finished.
+ */
+const char* gnc_relative_date_description(RelativeDatePeriod);
+
+/**
+ * Convert a relative date storage string back to a RelativeDatePeriod value.
+ *
+ * @param A string representation obtained from
+ * gnc_relative_date_storage_string.
+ * @return A RelativeDatePeriod value.
+ */
+RelativeDatePeriod gnc_relative_date_from_storage_string(const char*);
+
+/**
+ * Convert a RelativeDatePeriod value to a concrete time64 by applying the value to the current time. For example if it is now 3:15:42 PM local time 3 June, calling this with a period RelativeDatePeriod::ONE_WEEK_AHEAD will return a time64 representing 3:15:42 PM local time 10 June of this year.
```

Comment:
Perhaps also add an example for a start or end calculation illustrating that for those conversions the time part does change.

Reply:
OK.